framework/src/parser/MooseSyntax.C

```diff
@@ -62,12 +63,16 @@ associateSyntax(Syntax & syntax, ActionFactory & action_factory)
 
   /// Variable/AuxVariable Actions
   syntax.registerActionSyntax("AddVariableAction", "Variables/*");
+  syntax.registerSyntaxType("Variables/*", "VariableName");
+  syntax.registerSyntaxType("Variables/*", "NonlinearVariableName");
```

Comment:
I think there may be one or two places where we allow _any_ variable to be coupled in (Aux or NL). Can we handle that here?

Also, Here's another chunk of code that we should be able to clean up now or later:
https://github.com/idaholab/moose/blob/devel/framework/src/utils/InputParameters.C#L488-L558